app/config.js

```diff
@@ -233,3 +235,174 @@ export function newWalletConfigCreation(testnet, walletPath) {
     ini.stringify(dcrwConf)
   );
 }
+
+export function checkNoLegacyWalletConfig(testnet, walletPath, noLegacyRpc) {
+  let fileContents;
+  try {
+    fileContents = ini.parse(
+      fs.readFileSync(dcrwalletConf(getWalletPath(testnet, walletPath)), "utf8")
+    );
+    fileContents = Object.fromEntries(
+      Object.entries(fileContents).map(([key, value]) => {
+        if (key == "Application Options") {
+          if (value.nolegacyrpc && noLegacyRpc) {
+            delete value.nolegacyrpc;
+          } else if (!noLegacyRpc) {
+            value.nolegacyrpc = "1";
+          }
+        }
+        return [key, value];
+      })
+    );
+    fs.writeFileSync(
+      dcrwalletConf(getWalletPath(testnet, walletPath)),
+      ini.stringify(fileContents)
+    );
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+export const getDefaultBitcoinConfig = () =>
+  new Promise((resolve, reject) => {
+    try {
+      const config = ini.parse(
+        fs.readFileSync(
+          path.join(getDefaultBitcoinDirectory(), "bitcoin.conf"),
+          "utf8"
+        )
+      );
+      resolve(config);
+    } catch (e) {
+      reject(e);
+    }
+  });
+
+export function newDefaultBitcoinConfig(
+  rpcuser,
+  rpcpassword,
+  rpcbind,
+  rpcport,
+  testnet
+) {
+  if (!fs.existsSync(path.join(getDefaultBitcoinDirectory(), "bitcoin.conf"))) {
+    let bitcoinConf = {};
+    if (testnet) {
+      bitcoinConf = {
+        rpcuser,
+        rpcpassword,
+        server: 1,
+        test: {
+          rpcbind,
+          rpcport
+        }
+      };
+    } else {
+      bitcoinConf = {
+        rpcuser,
+        rpcpassword,
+        rpcbind,
+        rpcport,
+        server: 1
+      };
+    }
+    fs.writeFileSync(
+      path.join(getDefaultBitcoinDirectory(), "bitcoin.conf"),
+      ini.stringify(bitcoinConf)
+    );
+  }
+}
+
+export const updateDefaultBitcoinConfig = (
+  rpcuser,
+  rpcpassword,
+  rpcbind,
+  rpcport,
+  testnet
+) =>
+  new Promise((resolve, reject) => {
+    try {
+      // Check if default file exists, if not create a new one with args given.
+      if (
+        !fs.existsSync(path.join(getDefaultBitcoinDirectory(), "bitcoin.conf"))
+      ) {
+        // check to see if directory exists, if not make it
+        fs.pathExistsSync(getDefaultBitcoinDirectory()) ||
+          fs.mkdirsSync(getDefaultBitcoinDirectory());
+        newDefaultBitcoinConfig(
+          rpcuser,
+          rpcpassword,
+          rpcbind,
+          rpcport,
+          testnet
+        );
+      } else {
+        let fileContents;
+        let needUser = true;
+        let needPassword = true;
+        let needBind = true;
+        let needPort = true;
+        fileContents = ini.parse(
+          fs.readFileSync(
+            path.join(getDefaultBitcoinDirectory(), "bitcoin.conf"),
+            "utf8"
+          )
+        );
```

Comment:
This is the same as the `getDefaultBitcoinConfig, so could reuse the function if you change this (`updateDefaultBitcoinConfig`) to an async function (to make it easier to handle the various promises).